scripts/shared/cloud-prepare.sh

```diff
@@ -29,7 +29,58 @@ function check_gateway_exists() {
 
 function prepare_kind() {
     read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    kubectl label node "${cluster}-${nodes[-1]}" "${GATEWAY_LABEL}" --overwrite
+
+    # In order to test all gateway fail over test scenarios,
+    # one cluster will label 1 GW node and second 2 GW nodes.
+    worker_count=$(grep -o worker <<< "${nodes[*]}" | wc -l)
+
+    if [[ "$worker_count" -gt 1 ]]; then
+        echo "Multiple worker nodes detected - label two nodes as gateway"
+        worker_nodes=$(kubectl get nodes --no-headers=true \
+            -o custom-columns=NAME:.metadata.name | grep worker | head -n 2)
+
+        for node in ${worker_nodes}; do
+            kubectl label node "${node}" "${GATEWAY_LABEL}" --overwrite
+        done
+    else
+        echo "One worker node detected - label it as gateway"
+        kubectl label node "${cluster}-${nodes[-1]}" "${GATEWAY_LABEL}" --overwrite
+    fi
+
+    verify_gw_topology
+}
+
+# Verify that created clusters contains required gateway labeling
+# First cluster will contain 1 GW node
+# Second cluster will contain 2 GW nodes in active/passive mode
+function verify_gw_topology() {
```

Comment:
There's no reason to do it here, but you could add a dedicated test for it in the tests.

Reply:
@tpantelis explicitly asked me to add this verification to ensure the requested topology deployed.

Reply:
Moved near to "verify_gw_status" function